### PR TITLE
IEP-1100: Figuring out the status of OpenOCD process and then show some errors and refer to the documentation for OpenOCD

### PR DESCRIPTION
**Pull request by alirana01**
## Description

This will now show the FAQ link after any error is shown in the console for OpenOCD

Fixes # ([IEP-1100](https://jira.espressif.com:8443/browse/IEP-1100))

## Type of change

Please delete options that are not relevant.
- New feature (non-breaking change which adds functionality)

## How has this been tested?

Run a debug openocd configuration and try to create some errors a simple removal of the board should suffice as well, the following message along with URL for faq will be show

image

**Test Configuration**:
* ESP-IDF Version: ANY
* OS (Windows,Linux and macOS): ALL

## Checklist
- [x] PR Self Reviewed
- [x] Applied Code formatting
- [ ] Verified on all platforms - Windows,Linux and macOS

## Summary by CodeRabbit

- **New Features**
  - Enhanced the console output to detect and handle clickable URLs, improving user interaction with console logs.

- **Documentation**
  - Updated internal documentation to reflect new error and informational message handling related to the OpenOCD console.

- **Refactor**
  - Streamlined the addition of new pattern match listeners for future extensibility.

- **Chores**
  - Updated `.gitignore` to prevent unnecessary tracking of binary files.

### CODE REVIEW

**Review by alirana01 — COMMENTED** [review]
Self review

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/console/IdfProcessConsole.java [hunk 1]
```diff
@@ -186,4 +203,88 @@ public IOConsoleOutputStream getWarnStream()
...
+				String detectedUrl = idfProcessConsole.getDocument().get(offset, length);
+				IHyperlink link = new IHyperlink()
+				{
+					@Override
+					public void linkActivated()
+					{
+						try
+						{
+							Desktop.getDesktop().browse(new URI(detectedUrl));
+						}
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/console/IdfProcessConsole.java:240]: Can you use SWT based external browser functions for opening the url?

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/console/IdfProcessConsole.java [hunk 2]
```diff
@@ -4,7 +4,10 @@
  *******************************************************************************/
 package com.espressif.idf.debug.gdbjtag.openocd.dsf.console;
 
+import java.awt.Desktop;
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/console/IdfProcessConsole.java:7]: Use SWT browser

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/process/StreamListener.java [hunk 3]
```diff
@@ -19,30 +19,33 @@
...
+ * error and standard stream. This class is built using {@link ProcessConsole} StreamListner. The original class is
+ * private there.
+ * 
  * @author Ali Azam Rana
  */
 @SuppressWarnings("restriction")
 public class StreamListener implements IStreamListener
 {
-
+	private static final String OPENOCD_FAQ_LINK = "https://github.com/espressif/openocd-esp32/wiki/Troubleshooting-FAQ";
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/process/StreamListener.java:35]: Add non-NLS

**Review by kolipakakondal — APPROVED** [review]
LGTM

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/console/IdfProcessConsole.java (hunk 2 above)
Comment (alirana01) [reply bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/console/IdfProcessConsole.java]: Updated

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/process/StreamListener.java (hunk 3 above)
Comment (alirana01) [reply bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/process/StreamListener.java]: Added

**Comment by AndriiFilippov** [comment]
@alirana01 hi ! 


Tested under: 
OS - Windows 10 
ESP-IDF: v5.1.2 


Able to see active links in console during errors. 

image


LGTM 👍

**Review by AndriiFilippov — APPROVED** [review]

**Pull request merged**